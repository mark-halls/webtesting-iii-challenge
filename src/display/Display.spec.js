// Test away!
import React from "react";
import { render, cleanup } from "@testing-library/react";

import Display from "./Display";

afterEach(cleanup);

test("it renders Display", () => {
  const { getByText } = render(<Display closed={false} locked={false} />);
  expect(getByText(/open/i)).toMatchSnapshot();
  expect(getByText(/unlocked/i)).toMatchSnapshot();
});

test("it displays open/closed and if it is locked/unlocked", () => {
  const { getByText, rerender } = render(<Display closed={false} />);
  expect(getByText(/open/i));

  rerender(<Display closed={true} />);
  expect(getByText(/closed/i));

  rerender(<Display locked={true} />);
  expect(getByText(/locked/i));

  rerender(<Display locked={false} />);
  expect(getByText(/unlocked/i));
});

test("it uses the red-led class when locked or closed", () => {
  const { getByText, rerender } = render(<Display locked={true} />);
  getByText(/locked/i);
  expect(getByText(/locked/i).classList.contains("red-led")).toBe(true);

  rerender(<Display closed={true} />);
  getByText(/closed/i);
  expect(getByText(/closed/i).classList.contains("red-led")).toBe(true);
});

test("it uses the green-led class when unlocked or open", () => {
  const { getByText, rerender } = render(<Display locked={false} />);
  getByText(/lock/i);
  expect(getByText(/lock/i).classList.contains("green-led")).toBe(true);

  rerender(<Display closed={false} />);
  getByText(/open/i);
  expect(getByText(/open/i).classList.contains("green-led")).toBe(true);
});
