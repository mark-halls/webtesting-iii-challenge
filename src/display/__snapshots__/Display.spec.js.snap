// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders Display 1`] = `
<div
  class="led green-led"
>
  Open
</div>
`;

exports[`it renders Display 2`] = `
<div
  class="led green-led"
>
  Unlocked
</div>
`;
